### PR TITLE
Tls error

### PR DESCRIPTION
**Pull request by huitema**
I would like to add an error code, to use during the handshake when the TLS parameters proposed by the server cannot be accepted by the client. Example would be a bad certificate.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -3724,6 +3724,10 @@ PROTOCOL_VIOLATION (0xA):
 : An endpoint detected an error with protocol compliance that was not covered by
   more specific error codes.
 
+TLS_NEGOTIATION_FAILURE (0xB):
```
Comment (nibanks) [inline draft-ietf-quic-transport.md]: You need to fix `UNSOLICITED_PATH_RESPONSE (0xB)` below so there isn't collision on the error codes. That being said, what's wrong with `TLS_HANDSHAKE_FAILED` defined in the QUIC TLS spec?

Comment (huitema) [reply draft-ietf-quic-transport.md]: Nothing wrong with TLS_HANDSHAKE_FAILED. Apart from not being documented in the transport draft... I updated the PR to just a pointer to the TLS draft.

**Comment by huitema** [comment]
https://www.facebook.com/hatepseudoscience/photos/pb.163735987107605.-2207520000.1519251574./1023329751148220/

**Comment by martinthomson** [comment]
Love the subtle trolling in the image there :)

**Pull request merged**